### ISSUE TITLE
wish: open videofiles (mov, mp4, avi) with one shortkey (in VLC)

### ISSUE THREAD

**Issue by Cantuccini-Lori** [BestImageViewer/geeqie#728]
### ISSUE TYPE
Feature Request

### GEEQIE VERSION
Geeqie 1.5.1 in Debian Sid under Plasma 5.14

### OS / DISTRIBUTION
Debian Sid

### SUMMARY
As a photographer I only use open source, and I prefer geeqie because it is fast and works very well. But I cannot open videofiles. While its easy to open pics in gimp with a shortcut ("0" for me), I also need to control the videos and I never found a way to open them quickly in VLC.
So, is it possible to open videos in an external program (by plugin etc?) and give this a shortcut?
If there is already a way, it would be helpfull to know.
Thanks

**Comment by tomaszg7** [comment]
Sure there is! In Settings -> Behavior there is an option to play back videos just by clicking on them. You only need to set up plugin to handle them. 

Moreover when Geeqie is compiled with ffmpegthumbnailer support it renders thumbs for videos automatically. 

You might need to check filter settings to see if video files are enabled.

See also https://github.com/BestImageViewer/geeqie/issues/510

**Comment by Cantuccini-Lori** [comment]
Hi Tomaszg,
Thanks for your quick answer!
I had no time before Xmas, So I ansewer now. Hope, thats okay.
I found the behavior setting, but I need to define a player (play with), and it only shows imageprograms like darktable.
So I tired symolic link, but I really do not know how to define one.
Best would be for me to play within geeqie or with vlc.
What can I do?

**Comment by caclark** [comment]
Go to Edit/Configure Plugins/New
Change the Name line to be:
`Name=VLC`
Change the Exec line to be:
`Exec=vlc %f`
Save the plugin as vlc.desktop

Go to Edit/Preferences/Behavior.
You will now see that you can select "VLC" from the "Play with" drop-down.

Also go to Edit/Preferences/Keyboard.
Scroll down to the action VLC.
Click on the Key cell, and you can then set a keyboard shortcut.
You can use Help/Keyboard Map to select an unused keyboard shortcut.

**Comment by Cantuccini-Lori** [comment]
Thanks for helping. Unfortunatly it is not working. 

I went to "configure plugins/ New"
I got the white and empty windows and copied  

Name=VLC
Exec=vlc %f

into the Window, I named it "vlc.desktop", pressed save and close. Nut in the list with a lot of image formats "vlc" does not appear.
In "Edit/Preferences/Behavior" vlc also does not appear.

I tried also with a fresh config folder.
What did I wrong?

**Comment by caclark** [comment]
There may be a problem with the installation configuration. I will check that later.

For an immediate solution, open any other plugin file and copy the contents of that window into the blank window you get when pressing New. Change Name and Exec as appropriate before saving the new file.

**Comment by edward-dauvergne** [comment]
Why not simplify the user experience and cut back on options and just have an 'Enable external video playback' option?  That would enable all video extensions in the 'file types' list and enable the 'play video on left click' option.  Then a few builtin plugins for video players could be added with, for example, for VLC:

```
[Desktop Entry]
Version=1.0
Type=Application
Name=VLC
Exec=vlc --play-and-exit %F
```

The user could then choose the plugin, with 'Enable external video playback' being a combolist listing the plugins.  My distro is still on v1.4 so sorry for the noise if something has changed on HEAD.

**Comment by Cantuccini-Lori** [comment]
I still don't get it. I have Geeqie 1.5.1+git20200911-87c09755 installed.
I compilled with the debian script and have ffmpegthumbnail support (great!)

Then I went to Edit/Configure Plugins/New
Change the Name line to be:
Name=VLC
Change the Exec line to be:
Exec=vlc %f
Save the plugin as vlc.desktop

BUT: There is no possibility to find an vlc entry in Edit/Preferences/Behavior.
I cannot You select "VLC" from the "Play with" drop-down.

So I checked /usr/share/applications/ and found an old vlc.desktop file from august. I think this is from debian.
I do not know how to go on or where to look

**Comment by caclark** [comment]
Is there a vlc.desktop file in `$HOME/.config/geeqie/applications` ?
If you click `Plugins` on the main menu, does an entry for `vlc` show?
When you click the `Play With` option, do all the other plugins show?

**Comment by Cantuccini-Lori** [comment]
Thanks for your quick answer. I am not sure if this is the right place for troubleshooting, but I cannot find another place. So thanks for helping, seriously. I really like geeqie and miss only so few things.

so /home/USER/.config/geeqie/applications/
has the vlc.desktop file

in /usr/local/share/geeqie/applications/ are more (just to give the complete pic)
camera-import.desktop  
geeqie-ufraw-id.desktop         
random-image.desktop  
tethered-photography.desktop
export-jpeg.desktop    
geeqie-ufraw-recursive.desktop  
rotate.desktop        
geeqie-ufraw.desktop   
image-crop.desktop              
symlink.desktop

If I click on Plugins in the main menu a lot of plugins apperar (more than in the above loca/share!) but NO vlc.
If I click "play with" option a lot of plugins appear, as far As I see only picture plugins but NO Vlc

**Comment by caclark** [comment]
This is confusing. Perhaps you can go step-by-step through a procedure to isolate the problem.

For example (assuming camera-import is a plugin that is visible), execute:
`cp /usr/local/share/geeqie/applications/camera-import.desktop $HOME/.config/geeqie/applications/vlx.desktop`
Restart Geeqie (just to make sure the init. is done).
The plugins menu and the `play with` option should now show two entries with the name "camera import".
If that works, just change filename, name, exec, etc., one-by-one to isolate the problem.

**Comment by Cantuccini-Lori** [comment]
So, I did the following:
I copied the Camera-import to $HOME/.config/geeqie/applications/ and started geeqie
=> the pugin didn't appear in plugins

I made a copy of the camera plugin and renamed it to camera-import-.desktop and copied it into /usr/local/share/geeqie/applications/ and after restart I had two visble plugins "camera import"

I renamed the camera-import-.desktop to vlc.desktop and inserted
[Desktop Entry]
Version=1.0
Type=Application
Name=VLC
Exec=vlc --play-and-exit %F

And after a starting again, there was no new entry

Maybe it is the text that I inserted? And I agree, it's confusing. Until now, I cannot see a logic in it.

**Comment by caclark** [comment]
If you rename the desktop file from vlc.desktop to vlx.desktop you may find that it works.

It would seem there is a conflict between the desktop names - I will check that further tomorrow.

**Comment by Cantuccini-Lori** [comment]
unfortunatly nothing. also vlx.desktop ist not found in plugins

**Comment by Cantuccini-Lori** [comment]
Did you find anything? 
If I coould be of any help resolving the problem, please tell me :)

**Comment by caclark** [comment]
> I made a copy of the camera plugin and renamed it to camera-import-.desktop and copied it into /usr/local/share/geeqie/applications/ and after restart I had two visble plugins "camera import"

Go back to this step where the two plugins are displayed. This time change only one thing at a time before restarting. Which is the change that causes the problem.

---------------
Also, you could go to `Edit/Configure Plugins/New` , rename the item from `new.desktop`  to e.g. `caspita.desktop`.
Check if it displayed in the plugins list. If so, again, again just change one item at a time in the desktop file before checking if is ok or not ok.

It is very tedious to do this, but I do not see any other way of isolating the problem.

**Comment by Cantuccini-Lori** [comment]
Thanks for your answer. I tried the following:

1. I copied the camera-import plugin, renamed it to "camera-import plugin-" and it showed up in the plugins-menu and also in Edit/Configure Plugins.

I then closed geeqie, changed the file by entering:
Desktop Entry]
Version=1.0
Type=Application
Name=VLC
Exec=vlc --play-and-exit %F

And the plugin entry was gone. Is my syntax correct? 

2. Then I created a new plugin, renamed it to caspita.desktop and it apperaed as file in /home/h/.config/geeqie/applications/.
After a new start of geeqie it didn't appear neither in plugins-menu nor in /edit/configure plugins 

All this is very strange and it seems like a bug. Did you succed in creating a vlc plugin?

**Comment by caclark** [comment]
This command will list the known plugins, (but will not be much help):
`geeqie  --debug=2 | grep editor_command_parse`

This command will create a new desktop file:
`printf %b '[Desktop Entry]\nName=vlc-test\nExec=vlc --play-and-exit %F\nType=Application\nCategories=X-Geeqie;\n' >$HOME/.config/geeqie/applications/xyz.desktop`
and works for me....

**Comment by Cantuccini-Lori** [comment]
Yepp, the print command works. I found a vlc-test plugin, that opens VLC.
So I uuppose my Syntax was wrong.
Thanks a lot! I was looking for years for this feature. 

Maybe you could add plugins/shortcuts to vlc and/or other videoplayers to geeqie, since a lot of ppl use both, pic & videos. Since people use mobiles its a lot more videos!
So thanks again.
If I can do anything for you and geeqie feel free to as. What about an article in my free-software blog?

**Comment by netsuso** [comment]
In case anyone arrives here in the future, I had the same problem as @Cantuccini-Lori, and my plugin wouldn't appear in geeqie. After following the suggestions and doing some tests, I found that the line `Categories=X-Geeqie;` is mandatory for geeqie to detect it as a valid plugin. Without that line, it doesn't recognize it.

**Comment by caclark** [comment]
@netsuso 
When creating a new desktop file from within Geeqie, (Edit/Configure Plugins/New) the default settings include `Categories=X-Geeqie`.

Was that default data not there, or did you create the desktop file by other means?

**Comment by netsuso** [comment]
@caclark Thanks for answering!

I created it from geeqie, and indeed the `Categories=X-Geeqie;` line is included in the template. But it's well below, surrounded by other optional parameters. And it's commented as "Desktop files that are usable *only* in Geeqie should be marked like this", so it also looked optional to me and I removed it, as I wanted the file to be as simple as possible.

Maybe that line should be at the top, right below `Type=Application`, so it's clear that it's needed for the plugin to work
